### PR TITLE
Implement DISCL-254, action button for MovieContent

### PR DESCRIPTION
**Pull request by rdumusc**

### CODE REVIEW

File: src/core/ContentAction.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+void ContentAction::setIconChecked( const QString icon )
+{
+    if( icon == iconChecked_ )
+        return;
+
+    iconChecked_ = icon;
+    emit iconCheckedChanged();
+}
+
+void ContentAction::setCheckable(bool value)
```
Comment (tribal-tec) [inline src/core/ContentAction.cpp]: const

File: src/core/ContentActionsModel.cpp [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+    if( role == ROLE_ACTION )
+    {
+        QVariant variant;
+        variant.setValue( static_cast<QObject*>( actions_[ index_.row() ] ));
+        return variant;
+    }
+    return QVariant();
+}
+
+int ContentActionsModel::rowCount( const QModelIndex &parent_ ) const
```
Comment (tribal-tec) [inline src/core/ContentActionsModel.cpp]: wrong space before &

File: src/core/ContentActionsModel.h [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    Q_DISABLE_COPY( ContentActionsModel )
+
+public:
+    ContentActionsModel();
+    ~ContentActionsModel();
+
+    QVariant data( const QModelIndex &index, int role ) const override;
+    int rowCount( const QModelIndex& parent = QModelIndex( )) const override;
+
+//    bool insertRows( int row, int count, const QModelIndex& parent ) override;
```
Comment (tribal-tec) [inline src/core/ContentActionsModel.h]: WHAAAT?

File: src/core/MovieContent.cpp [hunk 4]
```diff
@@ -38,26 +38,35 @@
...
     , controlState_( STATE_LOOP )
 {
+    createPlayPauseButton();
 }
 
 MovieContent::MovieContent()
     : controlState_( STATE_LOOP )
 {
+    // Only when loading from xml, the actions need to be recreated.
+    if( actions_.rowCount() == 0 )
```
Comment (tribal-tec) [inline src/core/MovieContent.cpp]: do this from base class

**Comment by rdumusc** [comment]
Comments addressed

**Comment by tribal-tec** [comment]
+1

**Comment by rdumusc** [comment]
Strange bug with cppcheck and q_property on cscsviz:
ContentAction.cpp:42: warning: Member variable 'ContentAction::iconChanged' is not initialized in the constructor. etc... I can't find any reference to such problems...

**Comment by rdumusc** [comment]
Suppressed cppcheck warning and rebased, merging

**Pull request merged**